backend/api/office_hours/oh_section.py

```diff
@@ -0,0 +1,92 @@
+"""OH Section API
+
+This API is used to access OH section data."""
+
+from fastapi import APIRouter, Depends
+
+from backend.models.office_hours.oh_section import OfficeHoursSectionDraft
+from backend.models.office_hours.oh_section_details import OfficeHoursSectionDetails
+from ..authentication import registered_user
+from ...models import User
+
+
+__authors__ = ["Sadie Amato", "Bailey DeSouza"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+api = APIRouter(prefix="/api/office-hours/section")
+openapi_tags = {
+    "name": "Office Hours",
+    "description": "Office hours section functionality",
+}
+
+
+@api.post("", response_model=OfficeHoursSectionDetails, tags=["Office Hours"])
+def new_oh_section(
+    oh_section: OfficeHoursSectionDraft,
+    subject: User = Depends(registered_user),
+    oh_section_service: OfficeHoursSectionService = Depends(),
+) -> OfficeHoursSectionDetails:
+    """
+    Adds a new OH section to the database
+
+    Returns:
+        OfficeHoursSectionDetails: OH Section created
+    """
+    return oh_section_service.create(subject, oh_section)
+
+
+@api.get("/{term_id}", response_model=list[OfficeHoursSectionDetails], tags=["Office Hours"])
+def get_oh_sections_by_term_id(
+    term_id: str, subject: User = Depends(registered_user), oh_section_service: OfficeHoursSectionService = Depends()
+) -> list[OfficeHoursSectionDetails]:
+    """
+    Gets list of OH sections by term ID
+
+    Returns:
+        list[OfficeHoursSectionDetails]: OH sections within the given term
+    """
+    return oh_section_service.get_sections_by_term(subject, term_id)
+
+
+@api.get("", response_model=list[OfficeHoursSectionDetails], tags=["Office Hours"])
```

Comment:
Similar to the comment above about potentially specifying term as a specific category within our api path! Also, If we are trying to get Office Hours Section info based on a term id, we may need to have the id a part of the endpoint string. 

`"term/{term_id}"`

Let me know what you guys think!!

Reply:
I've updated this, thank you!!